lms/djangoapps/instructor/views/api.py

```diff
@@ -437,7 +444,9 @@ def register_and_enroll_students(request, course_id):  # pylint: disable=too-man
 
                     try:
                         with transaction.atomic():
-                            enrollment_obj = create_and_enroll_user(email, username, name, country, password, course_id)
+                            enrollment_obj = create_and_enroll_user(
+                                email, username, name, country, password, course_id, course_mode
+                            )
                             reason = 'Enrolling via csv upload'
```

Comment:
It seems like there is an opportunity to move some redundant code out of this workflow.  What do you think about relocating the "reason" and ManualEnrollmentAudit lines into create_and_enroll_user?